### PR TITLE
Add Sundar

### PR DESCRIPTION
**Pull request by Spandamn**

### CODE REVIEW

File: data/mods/ssb/moves.ts [hunk 1]
```diff
@@ -1001,6 +1001,29 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		name: "One For All: Full Cowl - 100%",
+		pp: 30,
+		priority: 0,
+		flags: {contact: 1, protect: 1},
+		onTryMovePriority: 100,
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'All Out Pummelling', target);
```
Comment (KrisXV) [inline data/mods/ssb/moves.ts]:
```suggestion
			this.add('-anim', source, 'All Out Pummeling', target);
```

File: data/mods/ssb/pokedex.ts [hunk 2]
```diff
@@ -78,4 +78,9 @@ export const BattlePokedex: {[k: string]: ModdedSpeciesData} = {
 		inherit: true,
 		abilities: {0: "The Numbers Game"},
 	},
+	// Sundar
+	charizardmegax: {
+		inherit: true,
+		abilities: {0: 'One For All'},
```
Comment (KrisXV) [inline data/mods/ssb/pokedex.ts]:
```suggestion
		abilities: {0: "One For All"},
```

File: data/mods/ssb/random-teams.ts [hunk 3]
```diff
@@ -180,6 +180,12 @@ export const ssbSets: SSBSets = {
 		signatureMove: 'Disconnect',
 		evs: {def: 4, spa: 252, spe: 252}, nature: 'Modest',
 	},
+	Sundar: {
+		species: 'Charizard', ability: 'Blaze', item: 'Charizardite X', gender: 'M',
+		moves: ['Earthquake', ['Double Edge', 'Flare Blitz'], 'Roost'],
+		signatureMove: 'One For All: Full Cowl - 100%',
+		evs: {atk: 252, spd: 4, spe: 252}, ivs: {spa: 0}, nature: 'Jolly',
```
Comment (KrisXV) [inline data/mods/ssb/random-teams.ts:187]: This is confusing since there’s absolutely 0 reason to have 0 SpA

File: data/mods/ssb/abilities.ts [hunk 4]
```diff
@@ -409,6 +409,21 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
...
+		desc: "This Pokemon's contact moves have their power multiplied by 1.3. If this Pokemon KOs its target with a recoil move,it recovers 25% of its max HP.",
+		shortDesc: "Tough Claws + recovers 25% of max HP when KOs a Pokemon with a recoil move.",
+		onBasePowerPriority: 21,
+		onBasePower(basePower, attacker, defender, move) {
+			if (move.flags['contact']) {
+				return this.chainModify([0x14CD, 0x1000]);
+			}
+		},
+		onSourceAfterFaint(length, target, source, effect) {
+			if (effect && effect.effectType === 'Move' && effect.recoil) this.heal(source.baseMaxhp / 4);
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts]:
```suggestion
			if (effect?.effectType === 'Move' && effect?.recoil) this.heal(source.baseMaxhp / 4);
```

File: data/mods/ssb/abilities.ts [hunk 5]
```diff
@@ -409,6 +409,21 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 			pokemon.side.addSideCondition('auroraveil');
 		},
 	},
+
+	// Sundar
+	oneforall: {
+		desc: "This Pokemon's contact moves have their power multiplied by 1.3. If this Pokemon KOs its target with a recoil move,it recovers 25% of its max HP.",
+		shortDesc: "Tough Claws + recovers 25% of max HP when KOs a Pokemon with a recoil move.",
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts]:
```suggestion
		shortDesc: "Tough Claws + recovers 25% max HP when foe is KOed with a recoil move.",
```

File: data/mods/ssb/abilities.ts [hunk 6]
```diff
@@ -409,6 +409,21 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 			pokemon.side.addSideCondition('auroraveil');
 		},
 	},
+
+	// Sundar
+	oneforall: {
+		desc: "This Pokemon's contact moves have their power multiplied by 1.3. If this Pokemon KOs its target with a recoil move,it recovers 25% of its max HP.",
```
Comment (KrisXV) [inline data/mods/ssb/abilities.ts]:
```suggestion
		desc: "This Pokemon's contact moves have their power multiplied by 1.3. If this Pokemon KOes the target with a recoil move, it regains 25% of its max HP.",
```

File: data/mods/ssb/random-teams.ts (hunk 3 above)
Comment (Spandamn) [reply data/mods/ssb/random-teams.ts:187]: well i put whatever's in the set

File: data/mods/ssb/moves.ts [hunk 7]
```diff
@@ -1001,6 +1001,29 @@ export const BattleMovedex: {[k: string]: ModdedMoveData} = {
...
+		priority: 0,
+		flags: {contact: 1, protect: 1},
+		onTryMovePriority: 100,
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'All Out Pummeling', target);
+		},
+		recoil: [40, 100],
```
Comment (ghost) [inline data/mods/ssb/moves.ts]:
```suggestion
		recoil: [4, 10],
```

**Pull request merged**